### ISSUE TITLE
bug: Duplicated peers in get_waku_v2_admin_v1_peers

### ISSUE THREAD

**Issue by AlbertoSoutullo** [waku-org/nwaku#1549]
### Problem
I don't know if it is actually a "bug" or not but didn't know what category to set.

If we have a node with store up for example, when calling `get_waku_v2_admin_v1_peers`, we get two results, one per protocol.

I don't know how this works in go-waku @richard-ramos 

### Impact
This can be confusing, as if you have 2 nodes, one connected to the other, you would expect one result.
This can also be confusing when plotting automatically, or doing tests (which was my case), as I was getting crazy about why I was getting 2 results instead one.

### To reproduce
1. Set up two wakunodes
2. Interconnect them
3. Do `get_waku_v2_admin_v1_peers` and you will get something similar to what I posted in example.

### Expected behavior
I don't know, I guess the expected result should be just one, and the protocols be grouped under the same key.

### Screenshots/logs
```
Request had response code '200' and body "{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[{\"multiaddr\":\"/ip4/0.0.0.0/tcp/60000/p2p/16Uiu2HAm8T4FHwZGcvo5YoBwbve6CLpWwgAgzCH29pMRbdCCR7VA\",\"protocol\":\"/vac/waku/relay/2.0.0\",\"connected\":true},{\"multiaddr\":\"/ip4/0.0.0.0/tcp/60000/p2p/16Uiu2HAm8T4FHwZGcvo5YoBwbve6CLpWwgAgzCH29pMRbdCCR7VA\",\"protocol\":\"/vac/waku/store/2.0.0-beta4\",\"connected\":true}]}\r\n"
```

Possible approach:
```
Request had response code '200' and body "{\"jsonrpc\":\"2.0\",\"id\":1,\"result\":[{\"multiaddr\":\"/ip4/0.0.0.0/tcp/60000/p2p/16Uiu2HAm8T4FHwZGcvo5YoBwbve6CLpWwgAgzCH29pMRbdCCR7VA\",\"protocols\": [\"/vac/waku/relay/2.0.0\", \"/vac/waku/store/2.0.0-beta4\"] ,\"connected\":true},
```

### nwaku version/commit hash
"statusteam/nim-waku:019f357d"

**Comment by alrevuelta** [comment]
Thanks @AlbertoSoutullo ! Yep, you are indeed right that this is missleading. You only have 1 peers but you get multiple entries, ome

**Comment by alrevuelta** [comment]
Thanks  @AlbertoSoutullo . Yep this is indeed misleading and I discussed it with @jm-clius that we need to rework this rpc call, but first we need to modify the spec and agree to a solution.

What is happening here is that this endpoint returns one `WakuPeer` entry per supported protocol. So if you are connected to just 1 peer, but this peer supports `relay` and `store`, then you will be returned 2 entries, almost equal but with the `protocol` field changed.

As a quick suggestion to be defined and tbd internally, I would suggest modifying [WakuPeer](https://github.com/waku-org/nwaku/blob/master/waku/v2/node/jsonrpc/admin/types.nim#L7-L10) to something like this. Where `identifyProtocols` is now an array that indicates the protocols that this peer supports (as per identify) and `protocolStreams` indicates the protocols that we are using for the connection(s).

```
type WakuPeer* = object
    multiaddr*: string
    identifyProtocols*: seq[string]
    protocolStreams*:seq[string]
    connected*: bool
```

Perhaps `get_waku_v2_admin_v1_peers` should be renamed to `get_waku_v2_admin_connected_peers` or something like that, to signify that its just returning connected peers.

tldr: you get one entry per peer per protocol. so 1 peer with 2 protocols, returns 2 entries. misleading imho.

**Comment by romanzac** [comment]
Hi, @alrevuelta @richard-ramos @AlbertoSoutullo Let me pull this one from ice box and update on actual status.

When two protocols (relay+store) are enabled on the node. These responses are received upon #get-/admin/v1/peers 

NWaku (v0.28.0):
```
Response status code: 200. Response content: b'[{"multiaddr":"/ip4/172.17.0.2/tcp/23049/p2p/16Uiu2HAmGNtM2rQ8abySFNhqPDFY4cmfAEpfo9Z9fD3NekoFR2ip","protocols":[{"protocol":"/vac/waku/store/2.0.0-beta4","connected":true}]}]'
```


GoWaku (latest 2024/6/4):
```
Response status code: 200. Response content: b'[{"id":"16Uiu2HAmQd2Ff6bdJ36pN5p79jiHLsJnvKz5U7D9ehZqgigoaVfP","multiaddrs":["/ip4/172.18.74.218/tcp/5875/p2p/16Uiu2HAmQd2Ff6bdJ36pN5p79jiHLsJnvKz5U7D9ehZqgigoaVfP","/ip4/172.18.74.218/tcp/5876/ws/p2p/16Uiu2HAmQd2Ff6bdJ36pN5p79jiHLsJnvKz5U7D9ehZqgigoaVfP"],"protocols":["/vac/waku/store-query/3.0.0","/vac/waku/store/2.0.0-beta4"],"connected":true,"pubsubTopics":[]}]'
```

Please refer to code at interop repo https://github.com/waku-org/waku-interop-tests/blob/873590e1fd9388c78e77922c600e480142f1e5f7/tests/peer_connection_management/test_peer_store.py#L59

Looking at variance of information provided by both implementations, I have some concerns about value for user/integrator. 

If you believe there is some value, I would personally prefer "lean & tested & frequently used" over "rich & somewhat tested & infrequently used". As long as we have Nim Waku centric strategy, I would pick nwaku to lead the change.

I would suggest to include simple array of peers [{'multiaddr': 'addr', 'protocol: proto'}] in the response. Although each multiaddr repeats the ID, it is unique because of port and related proto. From the service point of view, different port and proto actually point to a "different peer", even thou we decided for other reasons to run it on the same node.  Information wether peer is not connected looks somewhat redundant, as not connected peers are subject to reliability assessment for which we might have different API endpoints ? 

Let me know what you think. 

R.
